src/libcrun/status.c

```diff
@@ -45,6 +46,10 @@ get_run_directory (const char *state_root)
   if (root == NULL)
     root = xstrdup ("/run/crun");
 
+  ret = crun_ensure_directory (root, 0700, &err);
+  if (UNLIKELY (ret < 0))
```

Comment:
the function must return a `char *`, so we still need to return `root` here.

I've applied this patch on top of your PR and it works fine here:

```
diff --git a/src/libcrun/status.c b/src/libcrun/status.c
index c89ba16..50975b6 100644
--- a/src/libcrun/status.c
+++ b/src/libcrun/status.c
@@ -32,8 +32,9 @@
 static char *
 get_run_directory (const char *state_root)
 {
+  int ret;
   char *root = NULL;
-  libcrun_error_t err arg_unused;
+  libcrun_error_t err = NULL;
 
   if (state_root)
     root = xstrdup (state_root);
@@ -48,7 +49,7 @@ get_run_directory (const char *state_root)
 
   ret = crun_ensure_directory (root, 0700, &err);
   if (UNLIKELY (ret < 0))
-    return ret;
+    crun_error_release (&err);
```

Reply:
Ok updated and now I can actually compile it, and do a quick test.